### PR TITLE
Add support for TAN67 PCB

### PR DESCRIPTION
**Pull request by Pangorin**
## Description

Adding support for TAN67 PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by Pangorin — COMMENTED** [review]
Delete useless codes.

File: keyboards/pangorin/tan67/config.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* Bootmagic key */
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 0
```
Comment (zvecr) [inline keyboards/pangorin/tan67/config.h:21]:
```suggestion
```

File: keyboards/pangorin/tan67/keymaps/default/rules.mk
Comment (zvecr) [inline keyboards/pangorin/tan67/keymaps/default/rules.mk:1]: This file should be removed

File: keyboards/pangorin/tan67/readme.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# tan67 PCB
+
+![tan67 PCB]()
```
Comment (zvecr) [inline keyboards/pangorin/tan67/readme.md:3]: This should be updated to include an image, or the line removed.

File: keyboards/pangorin/tan67/tan67.c
Comment (zvecr) [inline keyboards/pangorin/tan67/tan67.c:1]: This file should be removed

File: keyboards/pangorin/tan67/rules.mk
Comment (zvecr) [inline keyboards/pangorin/tan67/rules.mk:1]: This should match the file <https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/rules.mk>.

File: keyboards/pangorin/tan67/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+└─────┘└─────┘└─────┘└─────────────────────────────────────┘└─────┘└─────┘  └────┘└────┘└────┘
+*/
+    /*  Row:    0        1        2        3        4        5        6        7        8        9        10       11       12       13       14     */
+    [_FN3]  = LAYOUT(
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+                _______, _______, _______,                   _______,                            _______, _______, _______, _______, _______
+            ),
```
Comment (zvecr) [inline keyboards/pangorin/tan67/keymaps/via/keymap.c:133]:
```suggestion
            )
```

File: keyboards/pangorin/tan67/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+└────┘└────┘└────┘└──────────────────────────────┘└────┘└────┘  └───┘└───┘└───┘
+*/
+    /*  Row:    0        1        2        3        4        5        6        7        8        9        10       11       12       13       14     */
+    [_FN3]  = LAYOUT(
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+                _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+                _______, _______, _______,                   _______,                            _______, _______, _______, _______, _______
+            ),
```
Comment (zvecr) [inline keyboards/pangorin/tan67/keymaps/default/keymap.c:133]:
```suggestion
            )
```

File: keyboards/pangorin/tan67/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum layer_names {
+    _BASE,
+    _FN1,
+    _FN2,
+    _FN3
```
Comment (zvecr) [inline keyboards/pangorin/tan67/keymaps/default/keymap.c:28]:
```suggestion
    _FN1
```

File: keyboards/pangorin/tan67/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum layer_names {
+    _BASE,
+    _FN1,
+    _FN2,
+    _FN3
```
Comment (zvecr) [inline keyboards/pangorin/tan67/keymaps/via/keymap.c:28]:
```suggestion
    _FN1
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/pangorin/tan67/keymaps/default/keymap.c (hunk 4 above)
Comment (Pangorin) [reply keyboards/pangorin/tan67/keymaps/default/keymap.c:133]: Can I keep this? (and in other keymaps). Because I want to put like 2 or 3 layers of Fn

**Review by Pangorin — COMMENTED** [review]
Update some files.

File: keyboards/pangorin/tan67/keymaps/default/keymap.c (hunk 5 above)
Comment (Pangorin) [reply keyboards/pangorin/tan67/keymaps/default/keymap.c:28]: I want to keep this too, I asked in another comment :D

**Review by Pangorin — COMMENTED** [review]
Edit rules.mk file.

File: keyboards/pangorin/tan67/rules.mk
Comment (Pangorin) [reply keyboards/pangorin/tan67/rules.mk:1]: I left it blank, but the PR Lint says it's falling. Is that normal?

**Review by Pangorin — COMMENTED** [review]
Fixing rules.mk file.

File: keyboards/pangorin/tan67/keymaps/default/keymap.c (hunk 4 above)
Comment (zvecr) [reply keyboards/pangorin/tan67/keymaps/default/keymap.c:133]: Nope, keymaps need to be minimal.

File: keyboards/pangorin/tan67/rules.mk
Comment (zvecr) [reply keyboards/pangorin/tan67/rules.mk:1]: It cannot be blank, it needs to match the file ive linked. Any lint/formatting issues will be due to not using unix line endings.

File: keyboards/pangorin/tan67/tan67.c
Comment (zvecr) [reply keyboards/pangorin/tan67/tan67.c:1]: This file still needs to be removed

File: keyboards/pangorin/tan67/keymaps/default/rules.mk
Comment (zvecr) [reply keyboards/pangorin/tan67/keymaps/default/rules.mk:1]: This file still needs to be removed

File: keyboards/pangorin/tan67/readme.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# tan67 PCB
```
Comment (HorrorTroll) [inline keyboards/pangorin/tan67/readme.md:1]: Should be uppercase `TAN67`

File: keyboards/pangorin/tan67/tan67.c
Comment (Pangorin) [reply keyboards/pangorin/tan67/tan67.c:1]: I added code to split out the functions of the underglow LED and the per-switch LED, so I could turn them on/off separately. Doesn't have to be the useless file anymore :D

File: keyboards/pangorin/tan67/readme.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+A 65% PCB, which controlled by an STM32F703 chipset.
```
Comment (HorrorTroll) [inline keyboards/pangorin/tan67/readme.md:3]: Grammar, it is `STM32F072`

File: keyboards/pangorin/tan67/readme.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+* Keyboard Maintainer: [Pangorin](https://github.com/pangorin)
+* Hardware Supported: STM32F703
```
Comment (HorrorTroll) [inline keyboards/pangorin/tan67/readme.md:6]: Same on line 3

File: keyboards/pangorin/tan67/readme.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+Flashing example for this keyboard:
+
+    make pangorin/tan67:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
```
Comment (HorrorTroll) [inline keyboards/pangorin/tan67/readme.md:20]: Wasn't it have only 2 ways?

**Review by HorrorTroll — CHANGES_REQUESTED** [review]

File: keyboards/pangorin/tan67/readme.md (hunk 10 above)
Comment (Pangorin) [reply keyboards/pangorin/tan67/readme.md:20]: Oh, I didn't notice that!

**Review by HorrorTroll — APPROVED** [review]

**Review by Pangorin — COMMENTED** [review]
Delete useless file, edit the keymap layers.

**Review by Pangorin — COMMENTED** [review]
More useless file deleted.

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by Pangorin — COMMENTED** [review]
Fixing errors.

**Comment by Pangorin** [comment]
@zvecr Sorry for the ping but I fixed errors :D

File: keyboards/pangorin/tan67/rules.mk [hunk 11]
```diff
@@ -0,0 +1 @@
+#This file intentionally left blank
```
Comment (zvecr) [inline keyboards/pangorin/tan67/rules.mk:1]:
```suggestion
# This file intentionally left blank
```

**Review by Pangorin — COMMENTED** [review]
Fixing typo.

**Review by Pangorin — COMMENTED** [review]
Fixing typo.

**Review by drashna — APPROVED** [review]

**Comment by Pangorin** [comment]
No review? .-.

**Comment by Pangorin** [comment]
I messed things up so I closed this pull request.

**Pull request closed without merging**